virtual_ecosystem/core/base_model.py

```diff
@@ -240,12 +241,110 @@ def __init__(
         """The core constants used in the model."""
         self._repr: list[tuple[str, ...]] = [("model_timing", "update_interval")]
         """A list of attributes to be included in the class __repr__ output"""
+        self._static = static
+        """Flag indicating if the model is static, i.e. does not change with time."""
+        self._run_update = self._run_update_due_to_static_configuration()
+        """Flag indicating if the update method should be run once.
+        
+        This is ignored if the model is not static."""
 
         # Check the required init variables
         self.check_init_data()
         # Check the configured update interval is within model bounds
         self._check_update_speed()
 
+        if not self._bypass_setup_due_to_static_configuration():
+            self._setup(**kwargs)
+
+    def _bypass_setup_due_to_static_configuration(self) -> bool:
```

Comment:
I'm a bit confused as to why setup needs to be bypassed (I'm sure it's something I'm missing). Why can't static models be set up in the same way as dynamic ones? Presumably the setup of static models happens before `__init__` in some way?